system/jlib/jfile.cpp

```diff
@@ -3105,8 +3105,10 @@ void CIOStreamReadWriteSeq::reset()
 
 size32_t read(IFileIO * in, offset_t pos, size32_t len, MemoryBuffer & buffer)
 {
-    const size32_t checkLengthLimit = 0x1000;
-    if (len >= checkLengthLimit)
+    constexpr size32_t checkLengthLimit = 0x1000;
```

Comment:
Change looks good.  Would it be sensible to add another 0 to the end of this?  64K would hardly be excessive these days...
I'll just check where this is actually called from....

Having looked, I think it may always be called with pos==0.  Could further clean up, and delete the pos parameter.

Reply:
yes, I see they are all pos==0 now.
But it's provided a IFileIO, a pos makes sense even if not used, I'm inclined to leave it in.

I wonder though, if it's ever intuitive  for it check size() if a size has been explicitly provided.
Maybe remove the limit check and check only wen len == -1 ?

Reply:
@ghalliday - I've made that change, please review/merge.